product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.1-Modify-payload-by-adding-a-child-using-enrich-mediator/src/test/java/org.wso2.carbon.ei.scenario.test/AddChildToXMLTest.java

```diff
@@ -89,6 +89,52 @@ public void testPassThroughProxyTemplate() throws IOException, XMLStreamExceptio
         assertResponse(response, expectedResponse);
     }
 
+    /**
+     * This test is to verify if payload can be modified by adding current payload
+     * as a child to the new payload
+     */
+    @Test(description = "1.6.1.3")
+    public void addCurrentPayloadAsChild() throws IOException, XMLStreamException {
+        String url = getProxyServiceURLHttp("1_6_1_3_Proxy_addCurrentPayloadAsChild");
+        String testCaseId = "1.6.1.3";
+        String request =
+                "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+                        + "<soap:Envelope xmlns:soap=\"http://schemas.xmlsoap.org/soap/envelope/\">\n"
+                        + "   <soap:Header />\n"
+                        + "   <soap:Body>\n"
+                        + "      <Employee>\n"
+                        + "         <firstName>Isuru</firstName>\n"
+                        + "         <lastName>Uyanage</lastName>\n"
+                        + "         <city>Colombo</city>\n"
+                        + "         <preferredName>Isuru</preferredName>\n"
+                        + "      </Employee>\n"
+                        + "   </soap:Body>\n"
+                        + "</soap:Envelope>";
+
+        String expectedResponse =
+                "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+                        + "<soap:Envelope xmlns:soap=\"http://schemas.xmlsoap.org/soap/envelope/\">\n"
+                        + "   <soap:Body>\n"
+                        + "      <Employee>\n"
+                        + "         <firstName>Isuru</firstName>\n"
+                        + "         <lastName>Uyanage</lastName>\n"
+                        + "         <city>Colombo</city>\n"
+                        + "         <preferredName>\n"
+                        + "            Isuru\n"
+                        + "            <Employee>\n"
```

Comment:
This use case is king of not standard since we do not append an element to text in this manner. let's replace entire content in the "preferredName" with new tag

Reply:
updated the proxy.